cylc/flow/task_job_mgr.py

```diff
@@ -308,18 +308,17 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
                     use_next_platform_in_group = False
                     bc_mgr = self.task_events_mgr.broadcast_mgr
                     rtconf = bc_mgr.get_updated_rtconfig(itask)
-                    if rtconf['platform']:
-                        try:
-                            platform = get_platform(
-                                rtconf['platform'],
-                                bad_hosts=self.bad_hosts
-                            )
-                        except PlatformLookupError:
-                            pass
-                        else:
-                            # If were able to select a new platform;
-                            if platform and platform != itask.platform:
-                                use_next_platform_in_group = True
+                    try:
+                        platform = get_platform(
+                            rtconf,
+                            bad_hosts=self.bad_hosts
+                        )
```

Comment:
Would be a good idea to check what other interfaces are using `get_platform(platform_name, ...)` as opposed to `get_platform(platform_config, ...)` to make sure this issue isn't present elsewhere.

Reply:
Searched `get_platform\([^\)]` and found the following cases:

- [x] `subprocpool.SubProcPool._run_command_exit` - Uses a platform name from a realized platform (i.e. `platform['name']` rather than from the config.
- [x] `task_events_mgr.TaskEventsManager._process_job_log_retrieval` - As above.
- [x] `task_job_mgr.TaskJobManager._run_job_cmd` - As above
- [x] `task_pool.TaskPool.load_db_task_pool_for_restart` - Loading a platform name from the database - which is set from `platform['name']`. 
- [x] `cat_log.main`  - as above
- [ ] `cylc check-versions` - is broken for compat mode tasks - shall I fix it here or create a new issue/pr - do people use it much?

Reply:
Great, thanks for checking.